std/mutex.zig

```diff
@@ -5,74 +5,145 @@ const AtomicRmwOp = builtin.AtomicRmwOp;
+                    _ = @atomicRmw(i32, &self.mutex.lock, AtomicRmwOp.Xchg, 0, AtomicOrder.Release);
+                    const rc = linux.futex_wake(&self.mutex.lock, linux.FUTEX_WAKE | linux.FUTEX_PRIVATE_FLAG, 1);
                     switch (linux.getErrno(rc)) {
                         0 => {},
                         linux.EINVAL => unreachable,
                         else => unreachable,
                     }
                 }
-            } else {
-                SpinLock.Held.release(SpinLock.Held{ .spinlock = &self.mutex.spin_lock });
             }
+        };
+
+        pub fn init() Mutex {
+            return Mutex {
+                .lock = 0,
+            };
         }
-    };
 
-    pub fn init() Mutex {
-        return Mutex{
-            .linux_lock = linux_lock_init,
-            .spin_lock = spin_lock_init,
-        };
-    }
+        pub fn deinit(self: *Mutex) void {}
 
-    pub fn acquire(self: *Mutex) Held {
-        if (builtin.os == builtin.Os.linux) {
-            var c = @cmpxchgWeak(i32, &self.linux_lock, 0, 1, AtomicOrder.Acquire, AtomicOrder.Monotonic) orelse
+        pub fn acquire(self: *Mutex) Held {
+            var c = @cmpxchgWeak(i32, &self.lock, 0, 1, AtomicOrder.Acquire, AtomicOrder.Monotonic) orelse
                 return Held{ .mutex = self };
             if (c != 2)
-                c = @atomicRmw(i32, &self.linux_lock, AtomicRmwOp.Xchg, 2, AtomicOrder.Acquire);
+                c = @atomicRmw(i32, &self.lock, AtomicRmwOp.Xchg, 2, AtomicOrder.Acquire);
             while (c != 0) {
-                const rc = linux.futex_wait(&self.linux_lock, linux.FUTEX_WAIT | linux.FUTEX_PRIVATE_FLAG, 2, null);
+                const rc = linux.futex_wait(&self.lock, linux.FUTEX_WAIT | linux.FUTEX_PRIVATE_FLAG, 2, null);
                 switch (linux.getErrno(rc)) {
                     0, linux.EINTR, linux.EAGAIN => {},
                     linux.EINVAL => unreachable,
                     else => unreachable,
                 }
-                c = @atomicRmw(i32, &self.linux_lock, AtomicRmwOp.Xchg, 2, AtomicOrder.Acquire);
+                c = @atomicRmw(i32, &self.lock, AtomicRmwOp.Xchg, 2, AtomicOrder.Acquire);
             }
-        } else {
-            _ = self.spin_lock.acquire();
+            return Held { .mutex = self };
         }
-        return Held{ .mutex = self };
-    }
+    },
+    builtin.Os.windows => struct {
+
+        lock: ?windows.CRITICAL_SECTION,
+        init_once: windows.PINIT_ONCE,
+
+        pub const Held = struct {
+            mutex: *Mutex,
+
+            pub fn release(self: Held) void {
+                if (self.mutex.lock) |*lock| {
+                    windows.LeaveCriticalSection(lock);
+                }
+            }
+        };
+
+        pub fn init() Mutex {
+            return Mutex { 
+                .lock = null,
+                .init_once = undefined,
```

Comment:
This needs to be initialized with INIT_ONCE_STATIC_INIT.

Reply:
`INIT_ONCE_STATIC_INIT` is a macro so i couldn't find it in any windows api bindings. so i looked through mingw-w64's sources and according to them, `INIT_ONCE_STATIC_INIT` is equal to [`RTL_RUN_ONCE_INIT`](https://github.com/mirror/mingw-w64/blob/master/mingw-w64-headers/include/synchapi.h#L19) and `RTL_RUN_ONCE_INIT` is equal to a [struct initializer](https://github.com/mirror/mingw-w64/blob/master/mingw-w64-headers/include/winnt.h#L7663) so i thought `undefined` would work the same in principe .

Reply:
This struct initializer means that all the fields are 0. So you need `INIT_ONCE{ .Ptr = null }`.